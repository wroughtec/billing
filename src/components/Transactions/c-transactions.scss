$tranSpacer: 2.5rem;

.c-transactions {
  padding-left: 0;
  margin-top: 0;
  margin-bottom: 0;

  list-style: none;
}

.c-transactions__item {
  display: grid;
  padding-top: .75rem;
  padding-bottom: .75rem;
  margin-right: $tranSpacer;
  margin-left: $tranSpacer;

  border-top: 1px solid var(--innerBorderColour);

  grid-template-columns: repeat(2, 50%);
}

.c-transactions__headings {
  border-top: 1px solid var(--innerBorderColour);
}

.c-transactions__title {
  padding-top: .75rem;
  margin-top: 0;
  margin-bottom: .5rem;
  margin-left: $tranSpacer;
}

.c-transactions__column-headings {
  display: grid;
  padding-right: $tranSpacer;
  padding-bottom: .75rem;
  padding-left: $tranSpacer;

  grid-template-columns: repeat(2, 50%);
}

.c-transactions__heading {
  font-weight: 700;
  text-transform: uppercase;
}

.c-transactions__info {
  color: var(--transactionColour);
}

.c-transactions__info--amount {
  margin-left: auto;
}
