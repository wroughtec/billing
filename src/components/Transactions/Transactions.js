// @flow

import React from 'react';
import Currency from 'react-currency-formatter';
import './c-transactions.scss';

type Props = {
  data: TransactionsType[]
};

export const Transactions = ({ data }: Props) => {
  let transactions = null;
  if (data && data.length) {
    const list = data.map(item => {
      const { id, date, amount } = item,
        transactionDate = new Intl.DateTimeFormat('en-GB').format(Date.parse(date));

      return (
        <li key={id} className="c-transactions__item">
          <span className="c-transactions__info">{transactionDate}</span>
          {amount && (
            <span className="c-transactions__info">
              <Currency quantity={amount} currency="GBP" />
            </span>
          )}
        </li>
      );
    });

    transactions = (
      <ul className="c-transactions">
        <li className="c-transactions__headings">
          <h4 className="c-transactions__title">Transactions</h4>
          <div className="c-transactions__column-headings">
            <span className="c-transactions__heading">Date</span>
            <span className="c-transactions__heading">Amount</span>
          </div>
        </li>
        {list}
      </ul>
    );
  }

  return transactions;
};
