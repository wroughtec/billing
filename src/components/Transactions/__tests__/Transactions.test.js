import React from 'react';
import { shallow } from 'enzyme';
import { Transactions } from '../Transactions';
import db from '../../../__mocks__/db.json';

describe('Transactions', () => {
  test('Renders', () => {
    const component = shallow(<Transactions data={db.bills[0].transactions} />);

    expect(component).toMatchSnapshot();
  });
});
