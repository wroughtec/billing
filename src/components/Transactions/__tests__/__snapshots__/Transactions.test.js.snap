// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions Renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Transactions
    data={
      Array [
        Object {
          "amount": 12.34,
          "date": "2018-01-13",
          "id": 36,
        },
        Object {
          "amount": 14.34,
          "date": "2018-02-13",
          "id": 37,
        },
        Object {
          "amount": 15.54,
          "date": "2018-03-13",
          "id": 38,
        },
        Object {
          "amount": 11.34,
          "date": "2018-04-13",
          "id": 39,
        },
        Object {
          "amount": 18.99,
          "date": "2018-05-13",
          "id": 40,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="c-transactions__headings"
        >
          <h4
            className="c-transactions__title"
          >
            Transactions
          </h4>
          <div
            className="c-transactions__column-headings"
          >
            <span
              className="c-transactions__heading"
            >
              Date
            </span>
            <span
              className="c-transactions__heading"
            >
              Amount
            </span>
          </div>
        </li>,
        Array [
          <li
            className="c-transactions__item"
          >
            <span
              className="c-transactions__info"
            >
              1/13/2018
            </span>
            <span
              className="c-transactions__info"
            >
              <ReactCurrencyFormatter
                currency="GBP"
                quantity={12.34}
              />
            </span>
          </li>,
          <li
            className="c-transactions__item"
          >
            <span
              className="c-transactions__info"
            >
              2/13/2018
            </span>
            <span
              className="c-transactions__info"
            >
              <ReactCurrencyFormatter
                currency="GBP"
                quantity={14.34}
              />
            </span>
          </li>,
          <li
            className="c-transactions__item"
          >
            <span
              className="c-transactions__info"
            >
              3/13/2018
            </span>
            <span
              className="c-transactions__info"
            >
              <ReactCurrencyFormatter
                currency="GBP"
                quantity={15.54}
              />
            </span>
          </li>,
          <li
            className="c-transactions__item"
          >
            <span
              className="c-transactions__info"
            >
              4/13/2018
            </span>
            <span
              className="c-transactions__info"
            >
              <ReactCurrencyFormatter
                currency="GBP"
                quantity={11.34}
              />
            </span>
          </li>,
          <li
            className="c-transactions__item"
          >
            <span
              className="c-transactions__info"
            >
              5/13/2018
            </span>
            <span
              className="c-transactions__info"
            >
              <ReactCurrencyFormatter
                currency="GBP"
                quantity={18.99}
              />
            </span>
          </li>,
        ],
      ],
      "className": "c-transactions",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="c-transactions__title"
            >
              Transactions
            </h4>,
            <div
              className="c-transactions__column-headings"
            >
              <span
                className="c-transactions__heading"
              >
                Date
              </span>
              <span
                className="c-transactions__heading"
              >
                Amount
              </span>
            </div>,
          ],
          "className": "c-transactions__headings",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Transactions",
              "className": "c-transactions__title",
            },
            "ref": null,
            "rendered": "Transactions",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="c-transactions__heading"
                >
                  Date
                </span>,
                <span
                  className="c-transactions__heading"
                >
                  Amount
                </span>,
              ],
              "className": "c-transactions__column-headings",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Date",
                  "className": "c-transactions__heading",
                },
                "ref": null,
                "rendered": "Date",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Amount",
                  "className": "c-transactions__heading",
                },
                "ref": null,
                "rendered": "Amount",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "36",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="c-transactions__info"
            >
              1/13/2018
            </span>,
            <span
              className="c-transactions__info"
            >
              <ReactCurrencyFormatter
                currency="GBP"
                quantity={12.34}
              />
            </span>,
          ],
          "className": "c-transactions__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1/13/2018",
              "className": "c-transactions__info",
            },
            "ref": null,
            "rendered": "1/13/2018",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ReactCurrencyFormatter
                currency="GBP"
                quantity={12.34}
              />,
              "className": "c-transactions__info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currency": "GBP",
                "quantity": 12.34,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "37",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="c-transactions__info"
            >
              2/13/2018
            </span>,
            <span
              className="c-transactions__info"
            >
              <ReactCurrencyFormatter
                currency="GBP"
                quantity={14.34}
              />
            </span>,
          ],
          "className": "c-transactions__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2/13/2018",
              "className": "c-transactions__info",
            },
            "ref": null,
            "rendered": "2/13/2018",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ReactCurrencyFormatter
                currency="GBP"
                quantity={14.34}
              />,
              "className": "c-transactions__info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currency": "GBP",
                "quantity": 14.34,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "38",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="c-transactions__info"
            >
              3/13/2018
            </span>,
            <span
              className="c-transactions__info"
            >
              <ReactCurrencyFormatter
                currency="GBP"
                quantity={15.54}
              />
            </span>,
          ],
          "className": "c-transactions__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "3/13/2018",
              "className": "c-transactions__info",
            },
            "ref": null,
            "rendered": "3/13/2018",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ReactCurrencyFormatter
                currency="GBP"
                quantity={15.54}
              />,
              "className": "c-transactions__info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currency": "GBP",
                "quantity": 15.54,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "39",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="c-transactions__info"
            >
              4/13/2018
            </span>,
            <span
              className="c-transactions__info"
            >
              <ReactCurrencyFormatter
                currency="GBP"
                quantity={11.34}
              />
            </span>,
          ],
          "className": "c-transactions__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "4/13/2018",
              "className": "c-transactions__info",
            },
            "ref": null,
            "rendered": "4/13/2018",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ReactCurrencyFormatter
                currency="GBP"
                quantity={11.34}
              />,
              "className": "c-transactions__info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currency": "GBP",
                "quantity": 11.34,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "40",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="c-transactions__info"
            >
              5/13/2018
            </span>,
            <span
              className="c-transactions__info"
            >
              <ReactCurrencyFormatter
                currency="GBP"
                quantity={18.99}
              />
            </span>,
          ],
          "className": "c-transactions__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5/13/2018",
              "className": "c-transactions__info",
            },
            "ref": null,
            "rendered": "5/13/2018",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ReactCurrencyFormatter
                currency="GBP"
                quantity={18.99}
              />,
              "className": "c-transactions__info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currency": "GBP",
                "quantity": 18.99,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
        ],
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="c-transactions__headings"
          >
            <h4
              className="c-transactions__title"
            >
              Transactions
            </h4>
            <div
              className="c-transactions__column-headings"
            >
              <span
                className="c-transactions__heading"
              >
                Date
              </span>
              <span
                className="c-transactions__heading"
              >
                Amount
              </span>
            </div>
          </li>,
          Array [
            <li
              className="c-transactions__item"
            >
              <span
                className="c-transactions__info"
              >
                1/13/2018
              </span>
              <span
                className="c-transactions__info"
              >
                <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={12.34}
                />
              </span>
            </li>,
            <li
              className="c-transactions__item"
            >
              <span
                className="c-transactions__info"
              >
                2/13/2018
              </span>
              <span
                className="c-transactions__info"
              >
                <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={14.34}
                />
              </span>
            </li>,
            <li
              className="c-transactions__item"
            >
              <span
                className="c-transactions__info"
              >
                3/13/2018
              </span>
              <span
                className="c-transactions__info"
              >
                <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={15.54}
                />
              </span>
            </li>,
            <li
              className="c-transactions__item"
            >
              <span
                className="c-transactions__info"
              >
                4/13/2018
              </span>
              <span
                className="c-transactions__info"
              >
                <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={11.34}
                />
              </span>
            </li>,
            <li
              className="c-transactions__item"
            >
              <span
                className="c-transactions__info"
              >
                5/13/2018
              </span>
              <span
                className="c-transactions__info"
              >
                <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={18.99}
                />
              </span>
            </li>,
          ],
        ],
        "className": "c-transactions",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="c-transactions__title"
              >
                Transactions
              </h4>,
              <div
                className="c-transactions__column-headings"
              >
                <span
                  className="c-transactions__heading"
                >
                  Date
                </span>
                <span
                  className="c-transactions__heading"
                >
                  Amount
                </span>
              </div>,
            ],
            "className": "c-transactions__headings",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Transactions",
                "className": "c-transactions__title",
              },
              "ref": null,
              "rendered": "Transactions",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="c-transactions__heading"
                  >
                    Date
                  </span>,
                  <span
                    className="c-transactions__heading"
                  >
                    Amount
                  </span>,
                ],
                "className": "c-transactions__column-headings",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date",
                    "className": "c-transactions__heading",
                  },
                  "ref": null,
                  "rendered": "Date",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Amount",
                    "className": "c-transactions__heading",
                  },
                  "ref": null,
                  "rendered": "Amount",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "36",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="c-transactions__info"
              >
                1/13/2018
              </span>,
              <span
                className="c-transactions__info"
              >
                <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={12.34}
                />
              </span>,
            ],
            "className": "c-transactions__item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1/13/2018",
                "className": "c-transactions__info",
              },
              "ref": null,
              "rendered": "1/13/2018",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={12.34}
                />,
                "className": "c-transactions__info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currency": "GBP",
                  "quantity": 12.34,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "37",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="c-transactions__info"
              >
                2/13/2018
              </span>,
              <span
                className="c-transactions__info"
              >
                <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={14.34}
                />
              </span>,
            ],
            "className": "c-transactions__item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2/13/2018",
                "className": "c-transactions__info",
              },
              "ref": null,
              "rendered": "2/13/2018",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={14.34}
                />,
                "className": "c-transactions__info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currency": "GBP",
                  "quantity": 14.34,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "38",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="c-transactions__info"
              >
                3/13/2018
              </span>,
              <span
                className="c-transactions__info"
              >
                <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={15.54}
                />
              </span>,
            ],
            "className": "c-transactions__item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "3/13/2018",
                "className": "c-transactions__info",
              },
              "ref": null,
              "rendered": "3/13/2018",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={15.54}
                />,
                "className": "c-transactions__info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currency": "GBP",
                  "quantity": 15.54,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "39",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="c-transactions__info"
              >
                4/13/2018
              </span>,
              <span
                className="c-transactions__info"
              >
                <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={11.34}
                />
              </span>,
            ],
            "className": "c-transactions__item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4/13/2018",
                "className": "c-transactions__info",
              },
              "ref": null,
              "rendered": "4/13/2018",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={11.34}
                />,
                "className": "c-transactions__info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currency": "GBP",
                  "quantity": 11.34,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "40",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="c-transactions__info"
              >
                5/13/2018
              </span>,
              <span
                className="c-transactions__info"
              >
                <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={18.99}
                />
              </span>,
            ],
            "className": "c-transactions__item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5/13/2018",
                "className": "c-transactions__info",
              },
              "ref": null,
              "rendered": "5/13/2018",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ReactCurrencyFormatter
                  currency="GBP"
                  quantity={18.99}
                />,
                "className": "c-transactions__info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currency": "GBP",
                  "quantity": 18.99,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
